vignettes/programming_strategy.Rmd

```diff
@@ -655,35 +653,49 @@ fun_xxx <- function(dataset, some_param, other_param) {
     other_param = other_param
 }
 ```
-**Phase 2:** In the next year/closest release in which a function is deprecated 
-the call to `deprecate_inform()` must be replaced with a call to `deprecate_warn()`.
+__NB:__ Please adjust phrase `{at the beginning of 20XX}` to relevant timeline.
+
+**Phase 2:** At the start of this phase the call to `deprecate_warn()` will 
+appear as:
 
 ```r
 fun_xxx <- function(dataset, some_param, other_param) {
-  deprecate_warn("x.y.z", "fun_xxx()", "new_fun_xxx()")
+  deprecate_warn(
+    when = "x.y.z", 
+    what = "fun_xxx()", 
+    with = "new_fun_xxx()",
+    details = c(
+              x = "This message will turn into a error {at the beginning of 20XX}.",
+              i = "See admiral's deprecation guidance:
+              https://pharmaverse.github.io/admiraldev/dev/articles/programming_strategy.html#deprecation"
+              )
   fun_xxx(
     dataset = dataset,
     some_param = some_param,
     other_param = other_param
   )
 }
 ```
+__NB:__ Please adjust phrase `{at the beginning of 20XX}` to relevant timeline.
 
-**Phase 3:** In the following release the function body should be changed to just include a call
-to `deprecate_stop()`.
+**Phase 3:** At the start of this phase the call to `deprecate_stop()` will 
+appear as:
 
 ```r
 fun_xxx <- function(dataset, some_param, other_param) {
-  deprecate_stop("x.y.z", "fun_xxx()", "new_fun_xxx()")
+  deprecate_stop(
+    when = "x.y.z", 
+    what = "fun_xxx()", 
+    with = "new_fun_xxx()",)
 }
 ```
 
-**Phase 4:** Finally, in the next release the function should be removed from the package.
+**Phase 4:** Function should be removed from the package.
 
 ### Argument
 
 **Phase 1:** If the argument is renamed or replaced, a **message** must be issued and the
-new argument takes the value of the old argument until the next release. Note:
+new argument takes the value of the old argument until the next phase. Note:
 arguments which are not passed as `exprs()` argument (e.g. `new_var = VAR1` or
 `filter = AVAL >10`) will need to be quoted.
 
```

Comment:
I would remove the `### BEGIN DEPRECATION` and `### END DEPRECATION` comments in the examples because it is inconsistent with the function deprecation and it contradicts our comment guidance.

Reply:
done!